ocean_lib/data_provider/data_service_provider.py

```diff
@@ -365,24 +384,39 @@ def build_endpoint(service_name, provider_uri=None, config=None):
         api_version = DataServiceProvider.get_api_version()
         if api_version not in provider_uri:
             provider_uri = f'{provider_uri}/{api_version}'
-
         return f'{provider_uri}/services/{service_name}'
 
     @staticmethod
     def build_encrypt_endpoint(provider_uri=None):
-        return DataServiceProvider.build_endpoint('encrypt', provider_uri)
+        for i in range(0, len(DataServiceProvider.valid_services)):
+            if DataServiceProvider.valid_services[i][0] == 'encrypt':
+                return DataServiceProvider.build_endpoint(DataServiceProvider.valid_services[i][0],
+                                                          DataServiceProvider.valid_services[i][1])
+        # return DataServiceProvider.build_endpoint('encrypt', provider_uri)
```

Comment:
we need to get the serviceEndpoints dict from the provider, using the root provider endpoint that was added in `https://github.com/oceanprotocol/provider/pull/46`. Then extract the (http-method, url) pair for the specific operation.
All the specific functions here should not be changed, the main change will be in `build_endpoint` function:
example:
```
def build_endpoint(service_name, provider_uri=None, config=None):
    ...
    endpoints_dict = DataServiceProvider.get_service_endpoints()
    method, url = endpoints_dict[service_name]
    return method, f'{provider_uri}/services/{url}'
```

Reply:
Ok, ok thank you for suggestion. Firstly, I modified in build_endpoint then I switched to all thos functions and I thounght it wasn't correct. I will refactor those.

Reply:
those* thought*

Reply:
Actually, it's not needed to have f'{provider_uri}/services/{url}' if I have the url from the endpoint_dict, because the format will add a redundant '/' and will duplicate the path. I think it should be f'{url}'.